charts/kubernikus-system/charts/prometheus/templates/_prometheus.yaml.tpl

```diff
@@ -241,7 +241,44 @@ scrape_configs:
       target_label: __metrics_path__
       replacement: /api/v1/nodes/${1}:4194/proxy/metrics
 
-# Static Targets
+- job_name: 'blackbox-ingress'
+  metrics_path: /probe
+  params:
+    # Look for a HTTP 200 response per default.
+    # Can be overwritten by annotating the ingress resource with the expected return codes, e.g. `prometheus.io/probe_code: "4xx"`
+    module: [http_2xx]
+  kubernetes_sd_configs:
+  - role: ingress
+  relabel_configs:
+  # don't scrape yourself
+  - source_labels: [__meta_kubernetes_ingress_path]
+    regex: /prometheus
+    action: drop
+  - source_labels: [__meta_kubernetes_ingress_annotation_prometheus_io_probe]
+    action: keep
+    regex: true
+  # consider prometheus.io/probe_code annotation. mind below regex.
+  - source_labels: [__meta_kubernetes_ingress_annotation_prometheus_io_probe_code]
+    regex: ^(\d).+
+    replacement: http_${1}xx
+    target_label: __param_module
+  - source_labels: [__meta_kubernetes_ingress_scheme,__address__,__meta_kubernetes_ingress_path]
+    regex: (.+);(.+);(.+)
+    replacement: ${1}://${2}${3}
+    target_label: __param_target
+  - target_label: __address__
+    replacement: blackbox-exporter.kubernikus-system.svc:9115
+  - source_labels: [__param_target]
+    target_label: instance
```

Comment:
It is my understanding that `instance` is a "reserved" label that get's filled by Prometheus. It is the instance where the metric emitted from. That would usually be the PodIP of the blackbox-exporter. 

If we had two black-box exporter pods, we couldn't distinguish between them. I'm unsure but wouldn't that even cause data "corruption" with both exporters reporting different values?

Reply:
We're using the ingress SD in this job, so the `instance` label is used for the discovered ingress and not for the blackbox exporter. The exporter itself gets discovered as a pod, where the `instance` label gets filled by prometheus

Reply:
Ah, ok. Makes sense.